File: 2265-partition-array-according-to-given-pivot/partition-array-according-to-given-pivot.py
class Solution:
    def pivotArray(self, nums: List[int], pivot: int) -> List[int]:
        smaller = []
        larger = []
        count_pivot = 0

        for num in nums:
            if num < pivot:
                smaller.append(num)
            elif pivot < num:
                larger.append(num)
            else:
                count_pivot += 1

        return smaller + [pivot] * count_pivot + larger
        

File: 2265-partition-array-according-to-given-pivot/partition-array-according-to-given-pivot.java
class Solution {
    public int[] pivotArray(int[] nums, int pivot) {
        int n = nums.length;
        int[] res = new int[n];
        int l = 0, r = n - 1;
        for (int i = 0; i < n; i++) {
            if (nums[i] < pivot) res[l++] = nums[i];
            if (nums[n - i - 1] > pivot) res[r--] = nums[n - i - 1];
        }

        while (l <= r) {
            res[l] = pivot;
            l++;
        }

        return res;
    }
}